ocsgw/src/main/java/org/ostelco/ocsgw/converter/ProtobufToDiameterConverter.java

```diff
@@ -162,6 +162,9 @@ private static void addPsInformation(final CreditControlContext context, CreditC
                 if (psInformation.getUserLocationInfo() != null) {
                     psInformationBuilder.setUserLocationInfo(ByteString.copyFrom(psInformation.getUserLocationInfo()));
                 }
+                if (psInformation.getPdpAddress() != null) {
+                    psInformationBuilder.setPdpAddress(psInformation.getPdpAddress().getHostAddress());
```

Comment:
Is this an FQDN or a raw IP address, if it's a raw IP address then it should be encoded somewhat differently. See the code proposed in "gyorde" (https://github.com/ExploratoryEngineering/gyorde/blob/master/protobuf/gyorde.proto) for a suggestion (handles both IPV4 and IPV6).

Reply:
This holds the IP address of the handset ( UE ).

I could encode this as:

message CheckDeviceRequest {
    enum IPType {
        IPV4 = 0;
        IPV6 = 1;
    }

    IPType ipType = 1;
    bytes ipAddress = 32;
};

If it matters if it is an IPv4 or v6 address.

Reply:
Added the full 3GPP-PDP-TYPE to the message. In our case we will always get this as IPv4, but now it is good for the future :)